src/js/stores/MesosStore.js

```diff
@@ -0,0 +1,247 @@
+var EventEmitter = require("events").EventEmitter;
+var semver = require("semver");
+
+var AppDispatcher = require("../AppDispatcher");
+var Util = require("../helpers/Util");
+var InfoStore = require("../stores/InfoStore");
+var InfoActions = require("../actions/InfoActions");
+var InfoEvents = require("../events/InfoEvents");
+var MesosActions = require("../actions/MesosActions");
+var MesosEvents = require("../events/MesosEvents");
+
+const FILES_TTL = 60000;
+const STATE_TTL = 60000;
+const REQUEST_DELAY = 100;
+const MASTER_ID = "master";
+
+var version = null;
+var stateMap = {};
+var taskFileMap = {};
+var taskFileRequestQueue = [];
+var requestCount = 0;
+
+function getDataFromMap(id, map, ttl = 100) {
+  if (!Util.isString(id) || map == null) {
+    return null;
+  }
+
+  let item = map[id];
+  if (item == null) {
+    return null;
+  }
+  if (item.timestamp + ttl < Date.now()) {
+    invalidateMapData(id, map);
+    return null;
+  }
+
+  return item.data;
+}
+
+function invalidateMapData(id, map) {
+  if (!Util.isString(id) || map == null) {
+    return;
+  }
+  delete map[id];
+}
+
+function addDataToMap(id, map, data) {
+  if (!Util.isString(id) || map == null || data == null) {
+    return;
+  }
+  map[id] = {
+    data: data,
+    timestamp: Date.now()
+  };
+}
+
+var MesosStore = Object.assign({
+
+  getState: function (nodeId) {
+    return getDataFromMap(nodeId, stateMap, STATE_TTL);
+  },
+
+  getTaskFiles: function (taskId) {
+    return getDataFromMap(taskId, taskFileMap, FILES_TTL);
+  }
+
+}, EventEmitter.prototype);
+
+function getNodeURLFromState(nodeId, state) {
+
+  if (state == null) {
+    return null;
+  }
+
+  let agent = state.slaves.find((slave) => {
+    return slave.id === nodeId;
+  });
+
+  if (agent == null) {
+    return null;
+  }
+
+  let pid = agent.pid;
+  let hostname = agent.hostname;
+  return `//${hostname}:${pid.substring(pid.lastIndexOf(":") + 1)}`;
+}
+
+function getExecutorDirectoryFromState(frameworkId, taskId, state) {
+
+  if (state == null) {
+    return null;
+  }
+
+  function matchFramework(framework) {
+    return frameworkId === framework.id;
+  }
+
+  let framework = state.frameworks.find(matchFramework) ||
+    state.completed_frameworks.find(matchFramework);
+
+  if (framework == null) {
+    return null;
+  }
+
+  function matchExecutor(executor) {
+    return taskId === executor.id;
+  }
+
+  let executor = framework.executors.find(matchExecutor) ||
+    framework.completed_executors.find(matchExecutor);
+
+  if (executor == null) {
+    return null;
+  }
+
+  return executor.directory;
+}
+
+function throttleRequest(request) {
```

Comment:
OK, got it! Thanx for the explanation.

Unfortunately it isn't still ideal.
![requests](https://cloud.githubusercontent.com/assets/859154/11657234/8e5ddce2-9dbb-11e5-9021-add0cdbf7e32.png)

I've got more then 10 requests on the info-endpoint in 5 seconds. 
That is not good.

I would suppose to discard the retry strategy.

Let's only try it once. If something in the request chain is failing, which won't happen that often, then we just display an error or something like that.
The user could reload the browser to try it again.